monai/data/itk_torch_affine_matrix_bridge.py

```diff
@@ -0,0 +1,350 @@
+):
+    """
+    Creates an affine transformation for an ITK image based on the provided parameters.
+
+    Args:
+        image: The ITK image.
+        translation: The translation (shift) to apply to the image.
+        rotation: The rotation to apply to the image, specified as angles in radians around the x, y, and z axes.
+        scale: The scaling factor to apply to the image.
+        shear: The shear to apply to the image.
+        center_of_rotation: The center of rotation for the image. If not specified,
+                            the center of the image is used.
+
+    Returns:
+        A tuple containing the affine transformation matrix and the translation vector.
+    """
+    itk_transform = itk.AffineTransform[itk.D, image.ndim].New()
+
+    # Set center
+    if center_of_rotation:
+        itk_transform.SetCenter(center_of_rotation)
+    else:
+        itk_transform.SetCenter(_get_itk_image_center(image))
+
+    # Set parameters
+    if rotation:
+        if image.ndim == 2:
+            itk_transform.Rotate2D(rotation[0])
+        else:
+            for i, angle_in_rads in enumerate(rotation):
+                if angle_in_rads != 0:
+                    axis = [0, 0, 0]
+                    axis[i] = 1
+                    itk_transform.Rotate3D(axis, angle_in_rads)
+
+    if scale:
+        itk_transform.Scale(scale)
+
+    if shear:
+        itk_transform.Shear(*shear)
+
+    if translation:
+        itk_transform.Translate(translation)
+
+    matrix = np.asarray(itk_transform.GetMatrix(), dtype=np.float64)
+
+    return matrix, translation
+
+
+def itk_affine_resample(image, matrix, translation, center_of_rotation=None):
+    # TODO documentation
+    # Translation transform
+    itk_transform = itk.AffineTransform[itk.D, image.ndim].New()
+
+    # Set center
+    if center_of_rotation:
+        itk_transform.SetCenter(center_of_rotation)
+    else:
+        itk_transform.SetCenter(_get_itk_image_center(image))
+
+    # Set matrix and translation
+    itk_transform.SetMatrix(itk.matrix_from_array(matrix))
+    itk_transform.Translate(translation)
+
+    # Interpolator
+    image = image.astype(itk.D)
+    interpolator = itk.LinearInterpolateImageFunction.New(image)
+
+    # Resample with ITK
+    output_image = itk.resample_image_filter(
+        image, interpolator=interpolator, transform=itk_transform, output_parameters_from_image=image
+    )
+
+    return np.asarray(output_image, dtype=np.float32)
+
+
+def monai_affine_resample(metatensor: MetaTensor, affine_matrix: NdarrayOrTensor):
+    # TODO documentation
+    affine = Affine(affine=affine_matrix, padding_mode="zeros", mode="bilinear", dtype=torch.float64, image_only=True)
+    output_tensor = cast(MetaTensor, affine(metatensor))
+
+    return cast(MetaTensor, output_tensor.squeeze().permute(*torch.arange(output_tensor.ndim - 2, -1, -1))).array
+
+
+def monai_to_itk_ddf(image, ddf):
```

Comment:
since this file now also includes some ddf related, the filename could be more generic... please also update the `__init__.py` to have these functions https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/__init__.py

Reply:
I don't know if it can help with this error but my idea with the functions `create_itk_affine_from_parameters', 'itk_affine_resample', 'monai_affine_resample', 'monai_wrap', 'itk_wrap' is that they should be part of the testing and not exposed to the user. So, maybe they could be moved to the test files and fix the cyclic imports?

Reply:
I'm sorry I'll move them. `remove_border` should still be exposed to the user, right?

Reply:
It was my fault to begin with of having everything in the same file ;) Also, `remove_border` is part of the testing so not exposed.

Reply:
for the circular import we usually avoid it by accessing the name later in the function:

```py
import monai

...
def function_that_uses_warp():
    ...
    x = monai.networks.blocks.Warp()
```